### ISSUE TITLE
BlockEditorView: Separate concerns of block definition format and editability

### ISSUE THREAD

**Issue by AnmAtAnm** [google/blockly-devtools#270]
(Copy of https://github.com/google/blockly/issues/1268)

Once https://github.com/google/blockly/pull/1235 is merged into devtools, block definition format selector will include "JSON", "JavaScript", and manual/editable versions of each. The "manual" options should probably be a state button to enable editing, separate from the format selector.
